src/jpgimage.cpp

```diff
@@ -273,14 +143,9 @@ void JpegBase::readMetadata() {
   byte marker = advanceToMarker(ErrorCode::kerNotAJpeg);
 
   while (marker != sos_ && marker != eoi_ && search > 0) {
-    // 2-byte buffer for reading the size.
-    std::array<byte, 2> sizebuf;
-    uint16_t size = 0;  // Size of the segment, including the 2-byte size field
-    if (markerHasLength(marker)) {
-      io_->readOrThrow(sizebuf.data(), sizebuf.size(), ErrorCode::kerFailedToReadImageData);
-      size = getUShort(sizebuf.data(), bigEndian);
-      enforce(size >= 2, ErrorCode::kerFailedToReadImageData);
-    }
+    std::array<byte, 2> sizebuf;  // 2-byte buffer for reading the size.
+    uint16_t size = 0;            // Size of the segment, including the 2-byte size field
+    readSegmentSize(marker, *io_, sizebuf, size);
```

Comment:
The inclusion of the new function `readSegmentSize` is also relevant. The same block of code was used in 3 different places and I factor it out to this new function.